Engine/Source/Runtime/ECWorld/ParticleEmitterComponent.h

```diff
@@ -161,11 +172,22 @@ class ParticleEmitterComponent final
 		cd::Vec4f color;
 		cd::UV     uv;
 	};
+
+	//Sprite
 	const cd::VertexFormat* m_pRequiredVertexFormat = nullptr;
-	std::vector<std::byte> m_particleVertexBuffer;
-	std::vector<std::byte> m_particleIndexBuffer;
-	uint16_t m_particleVertexBufferHandle = UINT16_MAX;
-	uint16_t m_particleIndexBufferHandle = UINT16_MAX;
+	std::vector<std::byte> m_spriteParticleVertexBuffer;
+	std::vector<std::byte> m_spriteParticleIndexBuffer;
+	uint16_t m_spriteParticleVertexBufferHandle = UINT16_MAX;
+	uint16_t m_spriteParticleIndexBufferHandle = UINT16_MAX;
+
+	//Ribbon
+	std::vector<std::byte> m_ribbonParticlePrePosVertexBuffer;
+	std::vector<std::byte> m_ribbonParticleRemainVertexBuffer;
+	std::vector<std::byte> m_ribbonParticleIndexBuffer;
+	uint16_t m_ribbonParticlePrePosVertexBufferHandle = UINT16_MAX;
+	uint16_t m_ribbonParticleRemainVertexBufferHandle = UINT16_MAX;
+	uint16_t m_ribbonParticleIndexBufferHandle = UINT16_MAX;
```

Comment:
Please think out a better solution after finish new feature. It is not good to keep Sprite and Ribbon data together. Because a particle emitter should not be able to emit two kinds of particles at the same time.